src/Filters/Block.php

```diff
@@ -964,4 +967,69 @@ private function preparePriceSpecifications()
             'currencySymbol' => $currency->sign,
         ];
     }
+
+    /**
+     * Get the reviews filter block
+     *
+     * @param array $filter
+     * @param array $selectedFilters
+     *
+     * @return array
+     */
+    private function getReviweBlock($filter, $selectedFilters)
+    {
+        $values = [
+            '4' => ['name'=>"4"],
+            '3' => ['name'=>"3"],
+            '2' =>  ['name'=>"2"],
+            '1' => ['name'=>"1"],
+        ];
+
+        $query = 'SELECT  count(g.grade) as count ,g.grade FROM (SELECT t.id_product, case 
+when t.avg_grade between 1 and 1.99 then "1"
+when t.avg_grade between 2 and 2.99 then "2"
+when t.avg_grade between 3 and 3.99 then "3"
+when t.avg_grade between 4 and 5 then "4"
+end as grade
+ FROM (SELECT avg(h.grade) as avg_grade,
```

Comment:
Two much subqueries, I'm afraid of performance problems :thinking:

Reply:
yes you are right, I think at least 3 joins are inevitable, but an alternative could be another table which keeps counts for all categories, and can be calculated during indexing.